cdap-docs/developers-manual/source/building-blocks/datasets/table.rst

```diff
@@ -207,3 +207,16 @@ When you want to delete all the columns of a row and you know all of them,
 passing all of them will make the deletion faster. Deleting all the columns of a row will
 also delete the entire row, as the underlying implementation of a Table is a 
 `columnar store. <http://en.wikipedia.org/wiki/Column-oriented_DBMS>`__
+
+.. _table-datasets-pre-splitting:
+
+Pre-Splitting a Table into Multiple Regions
+===========================================
+
+To create a Dataset pre-split into multiple regions, where the underlying storage is HBase,
+you can use::
```

Comment:
Where underlying storage for a Table Dataset (or any Dataset that uses Table underneath, e.g. KeyValueTable) is HBase, CDAP allows you to configure pre-splitting (see http://hbase.apache.org/book.html#manual_region_splitting_decisions) to gain better distribution of data operations after tables are created. This helps to optimize for better performance depending on your use case.

To specify splits for a Table based Dataset you will use "hbase.splits" dataset property. The value must contain JSON-formatted byte[][] of the split keys. E.g.:

```
"hbase.splits": "[[64],[128],[192]]"
```

The above will create 4 regions, first of which will get rows whose first byte is in the range 0..63, second will get 64..127, third will get 128..191 and fourth will get 192..255 ranges respectively.

You can set dataset properties when you create it with application deploy or via REST APIs. The following is the example of the former case. For the latter case, please refer to the Dataset section in REST API Reference (http://docs.cask.co/cdap/current/en/reference-manual/http-restful-api/dataset.html#creating-a-dataset).

To configure pre-splitting for the table created on application deploy, in your Application class' configure() method you can do:

``` java

public class MyApp extends AbstractApplication {

  @Override
  public void configure() {
    DatasetProperties props = DatasetProperties.builder().add("hbase.splits", "[[64],[128],[192]]").build();
    createDataset("myTable", KeyValueTable.class, props);

    // init other components
  }
} 

```